app/Main.hs

```diff
@@ -116,6 +118,18 @@ runServiceContainer stateVar (ServiceDefinition name imageName buildContext _ cr
       liftIO $ putStrLn $ "Starting " <> Text.unpack name <> "..."
       Docker.startContainer Docker.defaultStartOpts containerId
 
+ensureNetworking :: ServiceDefinition -> Docker.DockerT IO ()
+ensureNetworking (ServiceDefinition _ _ _ _ (Docker.CreateOpts _ _ networkingConfig)) =
+  mapM_ createNetworkWithName networkNames
```

Comment:
This can be useful, but maybe we should only create `hex-network` (always) to minimize hex's footprint and avoid conflicts with existing networks. Anyway, that's hardcoded now so I guess it's just handling the Docker endpoints config format here.

Something to think about but not a show-stopper 😸

Reply:
Good point about minimizing footprint, I also thought about creating network at the start only, but that option creates additional maintenance burden in the future - we need to remember that network is hardcoded here and there, and update both places accordingly. Current solution, while introducing a bit of overhead, is a no-brainer - if anything changes in how we process network configuration (for example, if we extend Hexfile format to handle that), it will be carried along automatically.